polkadot/runtime/westend/src/lib.rs

```diff
@@ -2255,6 +2258,54 @@ sp_api::impl_runtime_apis! {
 		}
 	}
 
+	impl xcm_fee_payment_runtime_api::XcmPaymentApi<Block, RuntimeCall> for Runtime {
```

Comment:
I'd move all of these implementations to a public function in pallet-xcm.
Then each of these functions can just call that public function.
It simplifies implementation of this API for all future runtimes.

Reply:
I added the implementations of the `query_weight_to_asset_fee` and the `query_delivery_fees` as public functions to pallet-xcm, as these functions will most certainly be implemented the same way in most chains.

However, I left the other two methods as is since different chains use different approaches to handling fee assets. It seems sensible to leave them to be implemented manually by a chain's team.